Neos.Neos/Classes/Domain/Pruning/ContentRepositoryPruningProcessor.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Neos.Neos package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Domain\Pruning;
+
+use Neos\ContentRepository\Core\ContentRepository;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryServiceInterface;
+use Neos\ContentRepository\Core\Feature\ContentStreamEventStreamName;
+use Neos\ContentRepository\Core\Feature\WorkspaceEventStreamName;
+use Neos\ContentRepository\Export\ProcessingContext;
+use Neos\ContentRepository\Export\ProcessorInterface;
+use Neos\EventStore\EventStoreInterface;
+
+/**
+ * Pruning processor that removes all events from the given cr
+ */
+final readonly class ContentRepositoryPruningProcessor implements ProcessorInterface, ContentRepositoryServiceInterface
+{
+    public function __construct(
+        private ContentRepository $contentRepository,
+        private EventStoreInterface $eventStore,
+    ) {
+    }
+
+    public function run(ProcessingContext $context): void
```

Comment:
this is a little in conflict with whats being done in: https://github.com/neos/neos-development-collection/pull/5297

eg id say we should still use the `ContentStreamPruner` here so we can also avoid this hardcore logic in neos

Reply:
Can we adjust that after #5297 is merged?

Reply:
Jup after the merge ill propose a change for this branch `feature/4448-site-import` how i would wire it together, but nothing at the logic will change.

Reply:
adjusted to use the `contentStreamPruner->pruneAllWorkspacesAndContentStreamsFromEventStream` here :) and no need to use the cr internals here as well ... as the cr needs to provide tools for this itself